gax-java/gax/src/main/java/com/google/api/gax/tracing/MetricsTracerFactory.java

```diff
@@ -45,13 +47,32 @@
 public class MetricsTracerFactory implements ApiTracerFactory {
   protected MetricsRecorder metricsRecorder;
 
+  // These are client attributes and pertain every single trace
+  private final Map<String, Object> attributes;
```

Comment:
I went back and forth between `Map<String, String>` and `Map<String, Object>`. Since `canUseDirectPath()` returns a boolean: https://github.com/googleapis/sdk-platform-java/blob/b1be846e2a830a19c9f9ad920336a63283b734cd/gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java#L466-L471, I would need to case it back to a String before adding it to the map. I was thinking that we accept whatever and then cast it ourselves with `String.valueOf()`.

Reply:
I think it is better to keep it consistent with the what we currently support in [MetricsTracer](https://github.com/googleapis/sdk-platform-java/blob/main/gax-java/gax/src/main/java/com/google/api/gax/tracing/MetricsTracer.java#L64). We could change the type in `MetricsTracer` to `Map<String, Object>` as well, but that mean more work in [OpenTelemetryMetricsRecorder](https://github.com/googleapis/sdk-platform-java/blob/main/gax-java/gax/src/main/java/com/google/api/gax/tracing/OpenTelemetryMetricsRecorder.java#L159-L164) because OpenTelemetry does not support put an attribute with `Object` value.

Reply:
Ok, I will revert back to Map<String, String>. Users will be expected to cast their value to a String themselves.